rpcserver.go

```diff
@@ -6570,3 +6600,26 @@ func encodeVirtualPackets(packets []*tappsbt.VPacket) ([][]byte, error) {
 
 	return rawPackets, nil
 }
+
+// EncodePrecisionInJSON takes a precision value and an encoded json object and
+// attempts to append a specific key-value pair in the json object, then encodes
+// it again and returns it.
+func EncodePrecisionInJSON(precision uint64, jBytes []byte) ([]byte, error) {
+	var jMeta map[string]interface{}
+
+	err := json.Unmarshal(jBytes, &jMeta)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal json asset meta: "+
+			"%w", err)
+	}
+
+	jMeta[MetadataPrecisionKey] = precision
```

Comment:
not sure if I should check for conflict here
it's not impossible that a user may have already defined this key (maybe intentionally, maybe by mistake :shrug: )

so perhaps on conflict maintain the user provided value? they *probably* know what they're doing
...or just error out

Reply:
I think returning an error instead of doing something implicit is probably better. Then the user has the option to just use the JSON or remove the value from the JSON and use the flag.